tests/unit/s2n_handshake_hashes_test.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include "tls/s2n_handshake_hashes.h"
+
+#include "tls/s2n_connection.h"
+
+/* Needed for s2n_handshake_get_hash_state_ptr */
+#include "tls/s2n_handshake.c"
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* Test s2n_handshake_hashes_new */
+    {
+        /* Safety */
+        EXPECT_ERROR_WITH_ERRNO(s2n_handshake_hashes_new(NULL), S2N_ERR_NULL);
+
+        /* Allocates a new s2n_handshake_hashes struct */
+        {
+            /* Allocates the struct */
+            struct s2n_connection conn = { 0 };
+            EXPECT_NULL(conn.handshake.hashes);
+            EXPECT_OK(s2n_handshake_hashes_new(&conn.handshake.hashes));
+            EXPECT_NOT_NULL(conn.handshake.hashes);
+
+            uint8_t data[100] = { 0 };
+
+            /* Allocates all hashes */
+            for (s2n_hash_algorithm alg = 0; alg < S2N_HASH_SENTINEL; alg++) {
+                if (alg == S2N_HASH_NONE) {
+                    continue;
+                }
+
+                uint8_t size = 0;
+                EXPECT_SUCCESS(s2n_hash_digest_size(alg, &size));
+                EXPECT_TRUE(size < sizeof(data));
```

Comment:
what is this testing?

Reply:
This isn't really testing anything. We just can't calculate a digest below without the right digest size :)